lib/pigeon/apns/config.ex

```diff
@@ -166,6 +166,23 @@ defimpl Pigeon.Configurable, for: Pigeon.APNS.Config do
 
   defdelegate close(config), to: Shared
 
+  def validate!(config) do
+    case config do
```

Comment:
It might be better to collect _all_ of the errors that are likely to happen here and raise a more complex error, rather than potentially having to play whack-a-mole with multiple configuration issues. I also think that the fact that the broken filename is missing will probably cause problems for debugging.

Reply:
I think a complex error could be raised, if every config item that failed is properly marked, and its failure properly explained. Otherwise, raising specific errors one by one is preferred.